app/test_main.py

```diff
@@ -1,3 +1,86 @@
+import pytest
 from app.main import get_human_age
 
-# write your code here
+
+@pytest.mark.parametrize(
+    "cat_age, dog_age, expected_years",
+    [
+        pytest.param(
+            -1, 8, [0, 0],
+            id="should return 0, 0"
+        ),
+        pytest.param(
+            2, -10, [0, 0],
+            id="should return 0, 0"
+        ),
+        pytest.param(
+            -2, -25, [0, 0],
+            id="should return 0, 0"
+        ),
+        pytest.param(
+            0, 0, [0, 0],
+            id="should return 0, 0"
+        ),
+        pytest.param(
+            14, 14, [0, 0],
+            id="should return 0, 0"
+        ),
+        pytest.param(
+            15, 15, [1, 1],
+            id="should return 1, 1"
+        ),
+        pytest.param(
+            23, 23, [1, 1],
+            id="should return 1, 1"
+        ),
+        pytest.param(
+            24, 24, [2, 2],
+            id="should return 2, 2"
+        ),
+        pytest.param(
+            27, 27, [2, 2],
+            id="should return 2, 2"
+        ),
+        pytest.param(
+            28, 28, [3, 2],
+            id="should return 3, 2"
+        ),
+        pytest.param(
+            100, 100, [21, 17],
+            id="should return 21, 17"
+        ),
+        pytest.param(
+            1.2, 8.5, [0, 0],
+            id="should return 0, 0"
+        ),
+        pytest.param(
+            5.1, 6.1, [0, 0],
+            id="should return 0, 0"
+        ),
+        pytest.param(
+            24.9, 25.6, [2, 2],
+            id="should return 2, 2"
+        ),
+        pytest.param(
+            27.1, 27.2, [2, 2],
+            id="should return 2, 2"
+        ),
+        pytest.param(
+            28.1, 28.7, [3, 2],
+            id="should return 3, 2"
+        ),
+        pytest.param(
+            100.3, 100.9, [21, 17],
+            id="should return 21, 17"
+        )
+    ]
+)
```

Comment:
You still have test duplicates. No need to add 3 tests for negative numbers, add only one for this case. The same in other cases, such as float numbers. Remove redundant tests and better add the check that 29 for a dog returns 3 in the result.